src/Microsoft.TemplateEngine.Cli/Commands/BaseCommand.cs

```diff
@@ -183,6 +183,25 @@ protected void SetupTabularOutputOptions(ITabularOutputCommand command)
             this.AddOption(command.ColumnsOption);
         }
 
+        protected void PrintDeprecationMessage<TDepr, TNew>(ParseResult parseResult, Option? additionalOption = null) where TDepr : Command where TNew : Command
+        {
+            var newCommandExample = Example.For<TNew>(parseResult);
+            if (additionalOption != null)
+            {
+                newCommandExample.WithOption(additionalOption);
+            }
+
+            Reporter.Output.WriteLine(string.Format(
+             LocalizableStrings.Commands_Warning_DeprecatedCommand,
+             Example.For<TDepr>(parseResult),
+             newCommandExample));
+
+            Reporter.Output.WriteLine(LocalizableStrings.Commands_Warning_DeprecatedCommand_Info);
+            Reporter.Output.WriteCommand(Example.For<TNew>(parseResult).WithHelpOption());
+            Reporter.Output.WriteLine();
```

Comment:
Do we have a mechanism for printing color? I'd expect warnings to be printed with a 'caution' color (yellow?) instead of plaintext (of course respecting NO_COLOR, TTY, etc).

Reply:
Is this a place where we'd need to get the IConsole off of the InvocationContext in order to have this kind of flexibility?

Reply:
yes, we do (however using `AnsiConsole` way); I will add color.
Worth to mention that not all the shells support color though.

`IConsole` didn't have this possibility out-of-box last time I checked.